src/scenes/AcceleratorRouter/MatrixView/index.tsx

```diff
@@ -87,36 +88,47 @@ const MatrixView = () => {
       },
     ];
 
-    const variableColumns: MRT_ColumnDef<ProjectsWithVariablesSearchResult>[] = uniqueVariableIds.map((variableId) => ({
-      id: variableId,
-      header: variableId,
-      size: 150,
-      accessorFn: (row) => {
-        const variableValue = row.variableValues.find((variable) => variable.variableId === variableId);
-
-        if (!variableValue) {
-          return '';
+    const variableNameMap = new Map<string, string>();
+    projects?.forEach((project) => {
+      project.variableValues?.forEach((variable) => {
+        if (!variableNameMap.has(variable.stableId)) {
+          variableNameMap.set(variable.stableId, variable.variableName);
         }
+      });
+    });
```

Comment:
Could we combine this with `uniqueVariableIds`? Then `uniqueVariableIds` just becomes the keys of the map